frontend/src/APIClients/lessonAPIClient.ts

```diff
@@ -0,0 +1,89 @@
+import baseAPIClient from "./BaseAPIClient";
+import AUTHENTICATED_USER_KEY from "../constants/AuthConstants";
+import { getLocalStorageObjProperty } from "../utils/LocalStorageUtils";
+
+/* eslint-disable @typescript-eslint/no-explicit-any */
+/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
+export type LessonRequest = {
+  course: string;
+  title: string;
+  description: string;
+  image: string;
+  content: [Record<string, any>];
+};
+
+export type LessonResponse = {
+  id: string | number;
+  course: string;
+  title: string;
+  description: string;
+  image: string;
+  content: [Record<string, any>];
+};
+
+// TO DO: error handling
+
+const getLesson = async (id: string): Promise<LessonResponse> => {
```

Comment:
Do we want any other getters? Off the top of my head,
- `get` -> returns all lessons
- `getLessonsByCourse` -> returns all lessons belonging to a course

Reply:
Currently, we only have the [getLessonById](https://github.com/uwblueprint/rowan-house/blob/main/backend/services/implementations/lessonService.ts#L15) in the back end. It's outlined in[ the ticket ](https://github.com/uwblueprint/rowan-house/issues/55)that retrieving all lessons of a module should be implemented separately
But if it makes sense, maybe this getter should be renamed to `getLessonById`.. lmk what you think!

Reply:
Meh it's pretty minor to me. Whatever is consistent/clear with other usages in repo